src/screens/Survey/redux/reducers/SurveyReducer.js

```diff
@@ -0,0 +1,100 @@
+import * as actionType from '../actions/ActionType';
+
+const initialState = {
+  index: 1,
+  maxIndex: 1,
+  totalNumQuestions: null,
+  questionResponses : [],
+  questionKeys: [],
+  databaseArgs: {
+    writeStatus: null,
+    errorType: null
+  }
+}
+
+const SurveyReducer = (state = initialState, action) => {
+    let newState;
+    let newIndex;
+    switch(action.type){
+      case actionType.LOAD_QUESTION_RESPONSE: {
+        let currentResponse = {questionId: action.questionId, response: action.response}
+        return newState = {
+          ...state,
+          questionResponses: [
```

Comment:
Would it be simpler to store these as a dictionary keyed by question id, so you didn't have to keep track of index?

Reply:
That's how I had it before, but there was the conversation about how arrays were better. Anyways, I have it back in the way it was. Hopefully that also helps with the other issue with match percentage